app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -651,7 +651,11 @@ public void find_new_raw_curve() {
             BgReading latest_entry = BgReading.lastNoSenssor();
             ra = 0;
             rb = 0;
-            rc = latest_entry.age_adjusted_raw_value;
+            if (latest_entry != null) {
```

Comment:
Syntactic testing is when I let the wixelReader.runFake actually run.

    public static BgReading lastNoSenssor() {
        return new Select()
                .from(BgReading.class)
                .where("calculated_value != 0")
                .where("raw_data != 0")
                .orderBy("timestamp desc")
                .executeSingle();
    }
last no sensor will not return anything with calculated_value = 0. and we have such packets in our db, before the first calibration. So, this seems to be the problem in my understanding.

see the code path of 
BgReading create(double raw_data, double filtered_data, Context context, Long timestamp) 
when sensor!=null and calibration == null.
Maybe we should set calculated_value there as well?